lighthouse-core/gather/connections/extension.js

```diff
@@ -99,25 +99,32 @@ class ExtensionConnection extends Connection {
 
   /**
    * @override
-   * @param {!string} method
+   * @param {!string} command
    * @param {!Object} params
    * @return {!Promise}
    */
   sendCommand(command, params) {
     return new Promise((resolve, reject) => {
       log.formatProtocol('method => browser', {method: command, params: params}, 'verbose');
       if (!this._tabId) {
-        log.error('No tabId set for sendCommand');
+        log.error('ExtensionConnection', 'No tabId set for sendCommand');
       }
+
       chrome.debugger.sendCommand({tabId: this._tabId}, command, params, result => {
         if (chrome.runtime.lastError) {
-          log.formatProtocol('method <= browser ERR', {method: command, params: result}, 'error');
-          return reject(chrome.runtime.lastError);
-        }
-
-        if (result.wasThrown) {
-          log.formatProtocol('method <= browser ERR', {method: command, params: result}, 'error');
-          return reject(result.exceptionDetails);
+          const message = chrome.runtime.lastError.message;
+          let error;
+          try {
+            error = JSON.parse(message);
+          } catch (e) {}
+          error = error || {message: 'Unknown debugger protocol error.'};
+
+          const callback = {
+            resolve,
```

Comment:
ok, the more I think about it, the more I like this pattern. I had avoided it in the past just because the `Promise.resolve() part feels like a waste (and the extra tick from it can be an issue for reasoning about things running in parallel), but the consistency in code is pretty nice. I'd be in favor of adopting this pattern in the future for functions returning a promise but also needing to swallow sync exceptions.

We need to be careful converting old code, though, because I know there are a few places we've written code explicitly assuming the executor will be executed synchronously (which is actually one more point in the consistency column if we always assume this pattern)